serde/src/lib.rs

```diff
@@ -86,6 +86,8 @@ mod core {
     pub use actual_core::nonzero;
 }
 
+extern crate void;
```

Comment:
I don't think we are getting enough bang for our buck out of this dependency. The entire functionality we need is provided equally well by:

```rust
enum Void {}
```

```rust
match self.void {}
```

If this were part of serde's public API then I would be all for using the void crate as a well-established way of signaling impossibility, but here it is just a private implementation detail...